docs/architecture/0003-replace-rdb-with-json.md

```diff
@@ -0,0 +1,65 @@
+
+Moreover, this check would likely need to happen on every request, resulting in increased complexity.
+
+Another motivating factor for revisiting our persistence mechanism is our current inability to easily port questionnaires between environments. 
+
+At present, there isn't an easy way of taking a questionnaire built in one environment e.g. pre-prod and migrating that data into another environment e.g. local/dev/test/prod. There are a few reasons for this, but one of the primary reasons is that database autogenerated Ids make it more difficult to ingest data from other environments and data would need to be migrated from many tables to do this successfully.
+
+### Motivations
+
+We hope making this change will make it:
+
+- Easier to make changes to the API going forward, less complexity.
+- Easier to write tests (most functionality can be tested by unit tests on pure functions).
+- Improvements to CI and Cypress test execution time - can load a document in a pre-configured state rather than having to create the state explicitly for each test.
+- Easier to move data between environments (pre-prod -> prod)
+- Easier to implement permissions checking.
+- Easier to keep versioned history of changes and restore to point in time.
+- Cheaper to run cloud infrastructure (serverless), and overall more performant.
+- More portable for if we need to migrate to another cloud provider.
+
+## Decision
+
+The decision taken is to refactor the API to use a JSON document database as a persistence mechanism.
+
+Initially our implementation will be focused on targeting DynamoDB, with the ability to run the application locally using either an in-memory or containerised local DynamoDB instance. However we're keen not to become dependent on any one cloud provider and so we should have a very thin layer of integration with Dynamo and rely on reading and mutating an in-memory JSON document for the most part.
+
+## Consequences
+
+### Benefits of making the change
+
+We believe that we will see benefits/improvements in the following areas after making this change:
+
+- Performance improvements (single read operation for queries, single write operation for mutations)
+- Simplified duplication logic.
+- Simplified ordering/moving logic.
+- Simplified author validation logic (we could leverage JSON schema validation for some of it).
+- Simplified code base (can probably delete most of the API code (repositories, data entities, migrations etc)
+- Since we would no longer on numeric autogenerated Ids from the database it will allow us to provide human-readable URLs which has been noted as being important for analytics.
```

Comment:
```suggestion
- Since we would no longer rely on numeric autogenerated Ids from the database it will allow us to provide human-readable URLs which has been noted as being important for analytics.
```